packages/core/src/render3/node_manipulation.ts

```diff
@@ -711,14 +711,15 @@ function getFirstNativeNode(lView: LView, tNode: TNode|null): RNode|null {
         }
       }
     } else {
-      const componentView = lView[DECLARATION_COMPONENT_VIEW];
-      const componentHost = componentView[T_HOST] as TElementNode;
-      const parentView = getLViewParent(componentView);
-      const firstProjectedTNode: TNode|null =
-          (componentHost.projection as (TNode | null)[])[tNode.projection as number];
-
-      if (firstProjectedTNode != null) {
-        return getFirstNativeNode(parentView!, firstProjectedTNode);
+      const projectionNodes = getProjectionNodes(lView, tNode);
+      if (projectionNodes !== null) {
+        if (Array.isArray(projectionNodes)) { 
+          return projectionNodes[0];
+        } else {
+          const parentView = getLViewParent(lView[DECLARATION_COMPONENT_VIEW]);
+          ngDevMode && assertParentView(parentView)
+          return getFirstNativeNode(parentView!, projectionNodes);
+        }
       } else {
         return getFirstNativeNode(lView, tNode.next);
```

Comment:
You can eliminate some `else` statements here.
```suggestion
        }
        const parentView = getLViewParent(lView[DECLARATION_COMPONENT_VIEW]);
        ngDevMode && assertParentView(parentView)
        return getFirstNativeNode(parentView!, projectionNodes);
      }
      return getFirstNativeNode(lView, tNode.next);
```

Reply:
Changed as requested.